scripts/multiqc

```diff
@@ -350,27 +350,32 @@ plots_flat, plots_interactive, make_pdf, config_file, cl_config, verbose, quiet,
                         modules_from_tags.add(module_name)
 
     # Get the list of modules we want to run, in the order that we want them
-    run_modules = [ m for m in config.top_modules if list(m.keys())[0] in config.avail_modules.keys() ]
-    run_modules.extend( [ {m:{}} for m in config.avail_modules.keys() if m not in mod_keys and m not in run_modules ] )
-    run_modules.extend( [ m for m in config.module_order if list(m.keys())[0] in config.avail_modules.keys() and list(m.keys())[0] not in [list(rm.keys())[0] for rm in run_modules] ] )
+    # config.run_modules gives us an OrderedDict to fill in
+    run_modules = config.run_modules
+    run_modules.update( [ (n, c) for m in config.top_modules for n, c in m.items() if n in config.avail_modules ] )
+    run_modules.update( [ (n, {}) for n in config.avail_modules if n not in mod_keys and n not in run_modules ] )
+    run_modules.update( [ (n, c) for m in config.module_order for n, c in m.items() if n in config.avail_modules and n not in run_modules ] )
 
     if module:
-        run_modules = [ m for m in run_modules if list(m.keys())[0] in module ]
-        logger.info('Only using modules {}'.format(', '.join(module)))
+        #Pare down config.run_modules
+        for m in list(run_modules):
+            if m not in module: del(run_modules[m])
```

Comment:
Does deleting a list item whilst looping over the same list like this work?

Reply:
Yes because the cast to `list()` does a shallow copy. If this was only for Py3 I would have done run_modules.copy() which is a little clearer.

Reply:
Gotcha 👍  Need to remember that trick! I usually end up writing iterators to do it, but they're not as clear to read.